mlflow/store/artifact/azure_blob_artifact_repo.py

```diff
@@ -38,10 +43,25 @@ def __init__(self, artifact_uri, client=None):
             self.client = BlobServiceClient(
                 account_url=account_url, credential=os.environ.get("AZURE_STORAGE_ACCESS_KEY")
             )
+        elif (
+            "AZURE_TENANT_ID" in os.environ
+            and "AZURE_CLIENT_ID" in os.environ
+            and "AZURE_CLIENT_SECRET" in os.environ
+        ):
+            from azure.identity import ClientSecretCredential
+
+            account_url = "https://{account}.blob.core.windows.net".format(account=account)
```

Comment:
The account URL construction here is only valid on the public cloud, but it's at least as correct as the current implementation - we can probably revisit this logic in a followup feature for private cloud support